config/lib/authorization.js

```diff
@@ -0,0 +1,91 @@
+﻿'use strict';
+
+var _ = require('lodash'),
+  config = require('../config'),
+  mongoose = require('mongoose'),
+  mg = require('./mongoose'),
+  jwt = require('jsonwebtoken'),
+  chalk = require('chalk');
+
+// build token auth service
+var tokenAuth = {
+  signToken: signAuthToken,
+  authorize: authorizeRequest
+};
+
+// export the token auth service
+module.exports = tokenAuth;
+
+// Sign a token
+function signAuthToken (user) {
+  return new Promise(function (resolve, reject) {
+    try {
+      var expiration = Date.now() + (config.tokenAuth.expiresInSeconds * 1000);
+
+      var payload = {
+        user: user._id,
+        expiration: expiration
+      };
+
+      var options = {
+        expiresIn: config.tokenAuth.expiresInSeconds
+      };
+
+      var tokenInfo = {
+        token: jwt.sign(payload, config.tokenAuth.secret, options),
+        expiration: expiration
```

Comment:
This method is creating a new signed Token. I use the user._id & expiration as the payload for generating the token; the expiration is beneficial since it is always going to be a unique value, thus it acts as a dynamic property for making the Token more secure.